script/generate-documentation

```diff
@@ -46,6 +46,23 @@
 # target remote ($1). Target branch will be created based on
 # PULL_REQUEST_USER:gh_pages
 
+set -eo pipefail
+
+# exit status 0 if any changes detected
+anything_changed() {
+    git --no-pager diff --shortstat
+    
+    local lines_changed_in_index lines_changed_in_api other_files_changed
```

Comment:
I personally prefer avoiding such lines and just use local on the first initialization.

Reply:
that I did as well, but shellcheck didn't like it for some reasons

Reply:
Ok, then let's keep it although I'm wondering why shellcheck enforces this.